src/types/gameConsole.ts

```diff
@@ -9,70 +9,103 @@ export default class GameConsole {
     // Fairchild
-    new GameConsole(/Channel F/i, [/* '.bin' */], 'channel_f', 'ChannelF'),
+    new GameConsole(/Channel F/i, [/* '.bin' */], 'channel_f', 'ChannelF', 'FAIRCHILD'),
+    // GCE
+    new GameConsole(/Vectrex/i, ['.vec'], undefined, undefined, 'VECTREX'),
     // Magnavox
-    new GameConsole(/Odyssey 2/i, [/* '.bin' */], 'odyssey2', 'Odyssey2'),
+    new GameConsole(/Odyssey 2/i, [/* '.bin' */], 'odyssey2', 'Odyssey2', 'ODYSSEY'),
     // Mattel
-    new GameConsole(/Intellivision/i, ['.int'], 'intv', 'Intellivision'),
+    new GameConsole(/Intellivision/i, ['.int'], 'intv', 'Intellivision', 'INTELLIVISION'),
+    // Microsoft
+    new GameConsole(/MSX/i, [], undefined, undefined, 'MSX'),
     // NEC
-    new GameConsole(/PC Engine|TurboGrafx/i, ['.pce'], 'pce', 'TGFX16'),
-    new GameConsole(/(PC Engine|TurboGrafx) CD/i, [/* '.bin', '.cue' */], 'pcecd', 'TGFX16'),
-    new GameConsole(/SuperGrafx/i, ['.sgx'], 'pce', 'TGFX16'),
+    new GameConsole(/PC Engine|TurboGrafx/i, ['.pce'], 'pce', 'TGFX16', 'PCE'),
+    new GameConsole(/(PC Engine|TurboGrafx) CD/i, [/* '.bin', '.cue' */], 'pcecd', 'TGFX16', 'PCECD'),
+    new GameConsole(/SuperGrafx/i, ['.sgx'], 'pce', 'TGFX16', 'SGFX'),
+    new GameConsole(/PC-88/i, ['.d88'], undefined, undefined, 'PCEIGHTYEIGHT'),
+    new GameConsole(/PC-98/i, ['.d98'], undefined, undefined, 'PCNINETYEIGHT'),
     // Nintendo
-    new GameConsole(/FDS|Famicom Computer Disk System/i, ['.fds'], 'nes', 'NES'),
-    new GameConsole(/GB|Game ?Boy/i, ['.gb', '.sgb'], 'gb', 'Gameboy'), // pocket:sgb for spiritualized1997
-    new GameConsole(/GBA|Game ?Boy Advance/i, ['.gba', '.srl'], 'gba', 'GBA'),
-    new GameConsole(/GBC|Game ?Boy Color/i, ['.gbc'], 'gbc', 'Gameboy'),
-    new GameConsole(/NES|Nintendo Entertainment System/i, ['.nes', '.nez'], 'nes', 'NES'),
-    new GameConsole(/Pokemon Mini/i, ['.min'], 'poke_mini', undefined),
-    new GameConsole(/Satellaview/i, ['.bs'], 'snes', 'SNES'),
-    new GameConsole(/SNES|Super Nintendo Entertainment System/i, ['.smc', '.sfc'], 'snes', 'SNES'),
+    new GameConsole(/FDS|Famicom Computer Disk System/i, ['.fds'], 'nes', 'NES', 'FDS'),
+    new GameConsole(/Game (and|&) Watch/i, ['.mgw'], undefined, undefined, 'GW'),
+    new GameConsole(/GB|Game ?Boy/i, ['.gb', '.sgb'], 'gb', 'Gameboy', 'GB'), // pocket:sgb for spiritualized1997
+    new GameConsole(/GBA|Game ?Boy Advance/i, ['.gba', '.srl'], 'gba', 'GBA', 'GBA'),
+    new GameConsole(/GBC|Game ?Boy Color/i, ['.gbc'], 'gbc', 'Gameboy', 'GBC'),
+    new GameConsole(/NES|Nintendo Entertainment System/i, ['.nes', '.nez'], 'nes', 'NES', 'FC'),
+    new GameConsole(/Pokemon Mini/i, ['.min'], 'poke_mini', undefined, 'POKE'),
+    new GameConsole(/Satellaview/i, ['.bs'], 'snes', 'SNES', 'SATELLAVIEW'),
+    new GameConsole(/Sufami/i, [], undefined, undefined, 'SUFAMI'),
+    new GameConsole(/SNES|Super Nintendo Entertainment System/i, ['.smc', '.sfc'], 'snes', 'SNES', 'SFC'),
```

Comment:
If you have a ROM in a folder titled `Nintendo - Super Nintendo Entertainment System`, it will match on `/NES|Nintendo Entertainment System/i` before it matches on `/SNES|Super Nintendo Entertainment System/i`

Reply:
It would, if the list didn't get reversed: https://github.com/emmercm/igir/blob/36eb1f83f868bcf2b8bfccdababaea20d8ac25bf/src/types/gameConsole.ts#L143

Game Boy and its sequels suffer from the same thing. I figure that Items appearing later in the list, if the list is alphabetized, will likely be more specific.

But I just found that `/NES/i` is matching `Benesse - Pocket Challenge V2` which is no good. I'm going to tweak this PR some more.

Reply:
Am I missing something? I'm not getting a `SFC` folder output, and am getting the SNES ROM put in the `FC` folder.
<img width="1017" alt="image" src="https://user-images.githubusercontent.com/2692529/236085143-d79dc4a4-0ce7-49a1-86a8-7df4570fd1a2.png">
<img width="665" alt="image" src="https://user-images.githubusercontent.com/2692529/236085064-9b70168d-923b-4f18-97d7-128f0029d3fe.png">

Reply:
What file extension are your SNES ROMs? I'm guessing the headered `.smc`? I had to remove the part where `igir` assumes `.smc` is always SNES, because `GamePark - GP32` uses that extension also.

But what I might do is add the extension back, and only consider extensions if DATs aren't provided.

I wish No-Intro provided a headered and headerless SNES DAT, like they do for NES.

Reply:
What's interesting, though, is with your No-Intro directory structure `igir` inferred a DAT name of `Nintendo - SUper Nintendo Entertainment System` so that should have taken priority.

Reply:
In case you're still looking for this answer, my SNES ROMs have the `.sfc` extension.

Reply:
This should fix it: https://github.com/emmercm/igir/pull/417